unlock-app/src/components/interface/locks/Manage/index.tsx

```diff
@@ -81,23 +82,23 @@ const TopActionBar = ({ lockAddress, network }: TopActionBarProps) => {
 
   const {
     register,
-    getValues,
     formState: { isValid, errors },
     resetField,
     setValue,
+    handleSubmit,
   } = useForm({
     mode: 'onChange',
     defaultValues: {
       url: '',
     },
   })
 
-  const { url } = getValues()
   const [isCopied, setCopied] = useClipboard(url || '', {
     successDuration: 2000,
   })
```

Comment:
This is using the redirect URI for the first click. The `url` comes from the useState which is the re-direct URI - that only changes after you click the form button once. 

store the generated url separately to pass into clipboard.